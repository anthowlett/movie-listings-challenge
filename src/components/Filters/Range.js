import React from 'react';
import PropTypes from 'prop-types';

const Range = ({
	value,
	min,
	max,
	step,
	onChange
}) => (
	<input type='range' min={min} max={max} step={step} value={value} onChange={(e) => {onChange(e.target.value)}} />
);

Range.propTypes = {
  value: PropTypes.number.isRequired,
  min: PropTypes.number,
  max: PropTypes.number,
  step: PropTypes.number,
  onChange: PropTypes.func.isRequired
};

Range.defaultProps = {
  min: 0,
  max: 10,
  step: 0.5
}

export default Range;