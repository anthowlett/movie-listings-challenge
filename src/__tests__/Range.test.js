import React from 'react';
import renderer from 'react-test-renderer';
import Range from '../components/Filters/Range';
import '../setupTests';

describe('Filters', () => {

  describe('<Range />', () => {
    const minRating = 3;
    const onSetMinRating = jest.fn();
    const component = <Range value={minRating} onChange={onSetMinRating} />
    const tree = renderer.create(component).toJSON();

    it('should render', () => {
      expect(tree)
        .toMatchSnapshot();
    });

    it('should not have child nodes', () => {
      expect(tree.children)
        .toBeFalsy();
    });
  });

});