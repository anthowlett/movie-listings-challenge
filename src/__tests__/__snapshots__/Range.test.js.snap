// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters <Range /> should render 1`] = `
<input
  max={10}
  min={0}
  onChange={[Function]}
  step={0.5}
  type="range"
  value={3}
/>
`;
